File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for i in s:
        if i == '(':
            stack.append(i)
        else: # 닫는 괄호가 나올 경우
            if stack == []: #스택이 비어있는데 닫는 괄호가 나오는 경우
                return False
            else:
                stack.pop()
    return stack == [] #스택이 비어있으면 true

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    let stack = []
    
    let arr_s = s.split("");
    if(arr_s[0] == ')'){
        return false;
    }
    arr_s.forEach((a) =>{
        // 여는 괄호일 떄
       if(a == '('){
           stack.push(a);
                       
       } 
        // 닫는 괄호일 때
        else{
          if(stack.length == 0){
              return false;
          }else{
              stack.pop();
          }
       }
    });
    
    if(stack.length == 0){
        return true;
    }
    return false;
}